### DISCUSSION TITLE
What does the bestmove (none) response mean actually?

### DISCUSSION THREAD

**Discussion by ghost** [official-stockfish/Stockfish#5075] [Q&A] [answered]
At ChesslaBlab, we've written the [Chess\UciEngine\Stockfish](https://github.com/chesslablab/php-chess/blob/main/src/UciEngine/Stockfish.php) class to allow PHP devs to interact with Stockfish. 

It seems to be working okay most of the time but sometimes, unexpectedly, its [bestMove()](https://github.com/chesslablab/php-chess/blob/main/src/UciEngine/Stockfish.php#L160) method won't work as expected because it has been designed under the assumption that Stockfish will always return a best move other than `(none)` no matter what the position is as long as there is a legal move to be made.

See https://github.com/chesslablab/chess-server/issues/256

Please note we're still developing this feature and our understanding of it may be not too accurate at the moment. Anyway, could you please confirm when Stockfish will respond with a `bestmove (none)` response?

Thanks for the help, and keep up with the great work!

**Comment by Disservin** [comment, marked as answer]
`bestmove (none)` is returned when Stockfish didn't find a bestmove, this can happen when the position is game over (i.e. checkmate) it should never happen when the position is legal and there's a legal move to be made.

Can you give us a log of when this is happening?


Some comments from me about your library or well your Stockfish class..

- make it independent of stockfish and let it work with all uci engines
- please don't hardcode options...you can extract all options from Stockfish after sending `uci`
https://github.com/chesslablab/php-chess/blob/main/src/UciEngine/Stockfish.php#L18
- i don't like your bestmove function... this is essentially restarting stockfish over and over again,
this will lose valuable information which stockfish has saved in the hash table and internal data structures... open the process in the constructor and close it in the destructor https://github.com/chesslablab/php-chess/blob/main/src/UciEngine/Stockfish.php#L160
- it's perhaps misleading to call this lan format, i never found this fitting, reported moves dont include the piece names https://github.com/chesslablab/php-chess/blob/main/src/UciEngine/Stockfish.php#L191, i.e. python chess makes a distinction https://python-chess.readthedocs.io/en/latest/core.html#chess.Move.uci
-  this evaluation is shit, it has no search associated with it, best if you don't expose this.. https://github.com/chesslablab/php-chess/blob/main/src/UciEngine/Stockfish.php#L216

**Comment by ghost** [comment]
Thanks for the response,

The log has recently been updated and doesn't contain any errors now. 

See https://github.com/chesslablab/php-chess/issues/482

Stay tuned!
